monai/bundle/reference_resolver.py

```diff
@@ -247,15 +247,17 @@ def find_refs_in_config(cls, config, id: str, refs: Optional[Set[str]] = None) -
             refs: list of the ID name of found references, default to `None`.
 
         """
-        refs_: Set[str] = refs or set()
+        refs_: List[str] = refs or []
         if isinstance(config, str):
-            return refs_.union(cls.match_refs_pattern(value=config))
+            for i in cls.match_refs_pattern(value=config):
+                if i not in refs_:
```

Comment:
It'll be nice to use a dictionary or a similar data structure to ensure unique references, better readability, and slightly better perf for the `in` checks.

Reply:
Sure, updated in the latest commit.
Could you please help review it again?

Thanks in advance.